electrum/lnwatcher.py

```diff
@@ -256,10 +265,18 @@ def maybe_add_pending_forceclose(
         spender_txid: Optional[str],
         is_local_ctx: bool,
         sweep_info: 'SweepInfo',
-        sweep_info_txo_is_nondust: bool,  # i.e. we want to sweep it
-    ):
-        """ we are waiting for ctx to be confirmed and there are received htlcs """
-        if is_local_ctx and sweep_info.name == 'received-htlc' and sweep_info_txo_is_nondust:
+    ) -> None:
+        """Adds chan into set of ongoing force-closures if the user should keep the wallet open, waiting for it.
+        (we are waiting for ctx to be confirmed and there are received htlcs)
+        """
+        if is_local_ctx and sweep_info.name == 'received-htlc':
+            cltv = sweep_info.cltv_abs
+            assert cltv is not None, f"missing cltv for {sweep_info}"
+            if self.adb.get_local_height() > cltv + REDEEM_AFTER_DOUBLE_SPENT_DELAY:
+                # We had plenty of time to sweep. The remote also had time to time out the htlc.
+                # Maybe its value has been ~dust at current and past fee levels (every time we checked).
+                # We should not keep warning the user forever.
+                return
```

Comment:
note: this is a functional change here.
Previously we only warned the user if the htlc was higher-than-dust value at *current* fee levels. Now that information is not readily available, so instead we always warn if there is an unspent received htlc. However, so that we don't keep warning forever, we now consider the cltv of the htlc. If it's past the cltv, the remote party could time it out at any point. Past the cltv, it's questionable if it makes sense to warn the user not to close the wallet.